lib/http/request.rb

```diff
@@ -106,7 +106,7 @@ def redirect(uri, verb = @verb)
 
     # Stream the request to a socket
     def stream(socket)
-      include_proxy_headers if using_proxy? && !@uri.https?
```

Comment:
Is there a reason this wouldn't have been enabled previously? I can't think of one off the top of my head.